src/plugins/ui_feature/insert_pivot.ts

```diff
@@ -169,34 +170,31 @@ export class InsertPivotPlugin extends UIPlugin {
     }
   }
 
-  splitPivotFormula(
-    sheetId: UID,
-    col: HeaderIndex,
-    row: HeaderIndex,
-    pivotId: UID,
-    pivotTableData: PivotTableData
-  ) {
-    this.dispatch("INSERT_PIVOT", {
-      sheetId,
-      col,
-      row,
-      pivotId,
-      table: pivotTableData,
-    });
+  splitPivotFormula(sheetId: UID, col: HeaderIndex, row: HeaderIndex, pivotId: UID) {
+    const spreadZone = this.getters.getSpreadZone({ sheetId, col, row });
+    if (!spreadZone) {
+      return;
+    }
+    const formulaId = this.getters.getPivotFormulaId(pivotId);
+
+    const pivotCells = new Map<CellPosition, PivotTableCell>();
+    for (let col = spreadZone.left; col <= spreadZone.right; col++) {
+      for (let row = spreadZone.top; row <= spreadZone.bottom; row++) {
+        const position = { sheetId, col, row };
+        pivotCells.set(position, this.getters.getPivotCellFromPosition(position));
```

Comment:
any reason accumulate everything in `pivotCells` to dispatch later rather than dispatching directly?

Reply:
I figured that updating the cells inside the spreaded zone would destroy the spreaded formula and make every subsequent call to `getPivotCellFromPosition` worn,g but maybe that only happens in the finalize ?

Reply:
It breaks my test :shrug: